.circleci/check-cpp-format.sh

```diff
@@ -0,0 +1,175 @@
+#!/bin/bash
+#
+# Copyright 2020-present Open Networking Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+set -ex
+
+# Files in this branch that are different from master.
+CHANGED_FILES=$(git diff-tree --no-commit-id --name-status -r HEAD origin/master \
+                  | grep -v "^D" \
+                  | cut -f 2 \
```

Comment:
git has a millions ways to do things, but this might be a little simpler:
```
git diff --name-only --diff-filter=d <BASE REF>
```

Reply:
This will return a list of all files that are not deleted between <BASE REF> and HEAD

Reply:
Actually we can get this whole thing down to a one-liner :)

```
git diff --name-only --diff-filter=d <BASE REF> -- '*.h' '*.cc'
```

Reply:
And git diff returns 0 whether or not it finds anything, so we can drop the `|| :`

Reply:
Amazing!